notebook/notebookapp.py

```diff
@@ -445,6 +450,10 @@ def _log_format_default(self):
         help="Set the Access-Control-Allow-Credentials: true header"
     )
     
+    allow_root = Bool(False, config=False, 
```

Comment:
You just have `config=False` here, which make it not configurable. Just switch to `config=True` and it works.